platform/ios/src/MGLMapView.mm

```diff
@@ -4106,7 +4106,7 @@ - (MGLAnnotationTag)annotationTagAtPoint:(CGPoint)point persistingResults:(BOOL)
             // The first selection in the cycle should be the one nearest to the
             // tap.
             CLLocationCoordinate2D currentCoordinate = [self convertPoint:point toCoordinateFromView:self];
-            std::sort(nearbyAnnotations.begin(), nearbyAnnotations.end(), [&](const MGLAnnotationTag tagA, const MGLAnnotationTag tagB) {
+            std::sort(_annotationsNearbyLastTap.begin(), _annotationsNearbyLastTap.end(), [&](const MGLAnnotationTag tagA, const MGLAnnotationTag tagB) {
```

Comment:
By design, this method doesn’t necessarily return the annotation nearest to the tap point. Instead, it’s intended to cycle among nearby annotations, starting with the nearest annotation. (This makes it possible for a user to access multiple overlapping annotations, even with their thumbs.) If this sort isn’t behaving as expected, we should figure out why.

The original code was intended to only affect `_annotationsNearbyLastTap` if `persist` is true. This changes the behavior to always affect `_annotationsNearbyLastTap`, so that preflight calls like the following cause the actual gesture recognizer to skip some of the annotations in the cycle:

https://github.com/mapbox/mapbox-gl-native/blob/8a283b030629abb00367929d047731c3974bc3d6/platform/ios/src/MGLMapView.mm#L1980-L1987

Reply:
Thanks for your review, I think the sort is behaving as expected. 
> The original code was intended to only affect _annotationsNearbyLastTap if persist is true. This changes the behavior to always affect _annotationsNearbyLastTap, so that preflight calls like the following cause the actual gesture recognizer to skip some of the annotations in the cycle

Yes, There is no need to affect `_annotationsNearbyLastTap` if `persist` is not true. I will change the code to sort temporary variable `nearbyAnnotations`.